src/main/flight/pid.c

```diff
@@ -1348,6 +1353,24 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
     float gyroRateDterm[XYZ_AXIS_COUNT];
     for (int axis = FD_ROLL; axis <= FD_YAW; ++axis) {
         gyroRateDterm[axis] = gyro.gyroADCf[axis];
+        // -----calculate raw, unfiltered D component
+
+        // Divide rate change by dT to get differential (ie dr/dt).
+        // dT is fixed and calculated from the target PID loop time
+        // This is done to avoid DTerm spikes that occur with dynamically
+        // calculated deltaT whenever another task causes the PID
+        // loop execution to be delayed.
+        const float delta =
+            - (gyroRateDterm[axis] - previousRawGyroRateDterm[axis]) * pidFrequency / D_LPF_RAW_SCALE;
+        previousRawGyroRateDterm[axis] = gyroRateDterm[axis];
+
+        // Log the unfiltered D
+        if (axis == FD_ROLL) {
+            DEBUG_SET(DEBUG_D_LPF, 0, lrintf(delta));
+        } else if (axis == FD_PITCH) {
+            DEBUG_SET(DEBUG_D_LPF, 1, lrintf(delta));
+        }
+
 #ifdef USE_RPM_FILTER
```

Comment:
The following few lines of filter application for the RPM filter, low pass filters and notch are all operating on the `gyro.gyroADCf[axis]` data, not the derivative. Are these then impacting D or the value of which D is the derivative. That's not what I'd believed these filters did until I looked at this code. The traces I show in this PR suggest that the RPM filter needs to be applied to the derivative. Thoughts?

Reply:
@SteveCEvans 
This is already implemented (using latest BF Configurator), this is the command in CLI: 
`set gyro_rpm_notch_q = 500`
Allowed range: 1 - 3000